rust-toolchain.toml

```diff
@@ -0,0 +1,2 @@
+[toolchain]
+channel = "nightly-2022-12-20"
```

Comment:
I think we can add the components too:

```toml
components = ["rustfmt", "clippy", "rust-src"]
```

Reply:
`rust-src` is not required for developing, `rustfmt` and `clippy` is enough I suppose

Reply:
The current behavior of with or without `components = ["rustfmt", "clippy"]` are the same, both use `default` profile, installed components are: `cargo clippy rust-docs rust-std rustc rustfmt`.

And we can get rid of `rust-docs` by using 
```
components = ["rustfmt", "clippy"]
profile = "minimal"
```

So: 
1. If we keep `rust-docs` installed, there's no need to add `components` block I think.
2. If we can remove `rust-docs`,  we should add `components` and `profile` block.

IMO we can choose 1 to keep it the same as before.

Reply:
WDYT ? @killme2008